<launch>
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video0" />
    <param name="camera_info_url" value="package://usb_cam/camera_info/head_camera.yaml"/>
    <param name="image_width" value="1280" />
    <param name="image_height" value="720" />
    <param name="pixel_format" value="mjpeg" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
  </node>

  <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
  <arg name="node_namespace" default="apriltags2_ros_continuous_node" />
  <!-- arg name="camera_name" default="/camera_rect" / -->
  <arg name="camera_name" default="/usb_cam" />
  <!-- arg name="camera_frame" default="camera" / -->
  <arg name="camera_frame" default="usb_cam" />
  <!-- arg name="image_topic" default="image_rect" / -->
  <arg name="image_topic" default="image_raw" />

  <!-- Set parameters -->
  <rosparam command="load" file="$(find apriltags2_ros)/config/settings.yaml" ns="$(arg node_namespace)" />
  <rosparam command="load" file="$(find apriltags2_ros)/config/tags.yaml" ns="$(arg node_namespace)" />
  <node pkg="apriltags2_ros" type="apriltags2_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
    <remap from="camera_info" to="$(arg camera_name)/camera_info" />

    <param name="camera_frame" type="str" value="$(arg camera_frame)" />
    <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
  </node>

  <!-- rviz -->
  <!-- node name="rviz" pkg="rviz" type="rviz" / -->

  <!-- tag_position -->
  <node name="tag_position" pkg="worker_pkg" type="tag_position.py" />

  <node name="image_view" pkg="image_view" type="image_view" output="screen">
   <param name="autosize" value="true" />
   <remap from="image" to="/tag_detections_image" />
  </node>

  <!-- joy -->
  <node name="joy_node" pkg="joy" type="joy_node" output="screen" >
    <param name="dev" value="/dev/input/js2" />
  </node>

  <node name="worker_control" pkg="worker_pkg" type="worker_control.py" output="screen" />

</launch>
